### PR TITLE
Log changes

### PR DESCRIPTION
**Pull request by vrahane**
- Adding/Changing fields(msg, ts, entries) to log entries
- Fixing json encode for 64 bit data types
- Adding os_time function to get uptime in microsenconds
- Changing json responses to contain multiple log entries

### CODE REVIEW

File: libs/os/src/os_time.c [hunk 1]
```diff
@@ -141,3 +141,19 @@ os_gettimeofday(struct os_timeval *tv, struct os_timezone *tz)
 
     return (0);
 }
+
+int64_t
+os_get_uptime_usec(void) {
```
Comment (sterlinghughes) [inline libs/os/src/os_time.c]: the { should be on the next line.

File: libs/os/src/os_time.c [hunk 2]
```diff
@@ -141,3 +141,19 @@ os_gettimeofday(struct os_timeval *tv, struct os_timezone *tz)
...
+  struct os_timeval tv;
+  os_time_t delta;
+  os_sr_t sr;
+
+  OS_ENTER_CRITICAL(sr);
+
+  delta = os_time_get() - basetod.ostime;
+  os_deltatime(delta, &basetod.uptime, &tv);
```
Comment (sterlinghughes) [inline libs/os/src/os_time.c]: This is a long period to disable a critical section.  can you just grab basetod.os_time, then do the calculation?

File: sys/log/include/log/log.h [hunk 3]
```diff
@@ -51,8 +51,9 @@ struct log_handler {
 
 struct log_entry_hdr {
     int64_t ue_ts;
-    uint16_t ue_level;
     uint16_t ue_module;
+    uint16_t ue_index;
+    uint8_t ue_level;
```
Comment (sterlinghughes) [inline sys/log/include/log/log.h]: This is unaligning the structure, please align the structure.  you can add a ue_arg0, or unused field if you want to, but the behavior should be defined here.

File: libs/baselibc/src/tinyprintf.c [hunk 4]
```diff
@@ -103,15 +103,15 @@ static void li2a(long num, struct param *p)
 }
```
Comment (sterlinghughes) [inline libs/baselibc/src/tinyprintf.c]: In general, please remove whitespace changes from patch submissions.  Once you get the commit bit, you can make those changes, but its hard to read through the patches and distinguish what is WS and what isn't -- and those WS changes are probably not that critical.

File: sys/log/src/log_nmgr.c [hunk 5]
```diff
@@ -76,17 +76,30 @@ log_nmgr_add_entry(struct log *log, void *arg, void *dptr, uint16_t len)
...
 
+    JSON_VALUE_UINT(&jv, ueh.ue_level);
+    rc = json_encode_object_entry(encoder, "level", &jv);
+    if (rc != 0) {
+        goto err;
+    }
+
+    /* Dummy index value for now */
+    JSON_VALUE_UINT(&jv, ueh.ue_index);
+    rc = json_encode_object_entry(encoder, "index", &jv);
```
Comment (sterlinghughes) [inline sys/log/src/log_nmgr.c]: Why is the index in the log, if its only a dummy index value?

File: libs/os/src/os_time.c (hunk 1 above)
Comment (vrahane) [inline libs/os/src/os_time.c]: Will fix this.

File: libs/os/src/os_time.c (hunk 2 above)
Comment (vrahane) [inline libs/os/src/os_time.c]: Will fix this. I can move os_deltatime() outside the critical section.

File: sys/log/include/log/log.h (hunk 3 above)
Comment (vrahane) [inline sys/log/include/log/log.h]: Fix is to have a log_level and index be a byte each. The problem is also that the alignment is going to be 8 bytes and so the sizeof the structure is going to be 16 bytes. I am going to pack it sop that it uses less space as this is going to go on the flash.

File: libs/baselibc/src/tinyprintf.c (hunk 4 above)
Comment (vrahane) [inline libs/baselibc/src/tinyprintf.c]: Point taken.

File: sys/log/src/log_nmgr.c (hunk 5 above)
Comment (vrahane) [inline sys/log/src/log_nmgr.c]: It is for the ordering for events that happen in a millisecond. I am changing it to be a byte as per the discussion and suggestion.

**Comment by vrahane** [comment]
Changing as per review comments.

**Pull request closed without merging**